gems/pending/spec/support/test_env_helper.rb

```diff
@@ -0,0 +1,177 @@
+  GLOBAL_CONFIG_FILE = File.join(TEST_ENV_DIR, 'config.yml')
+
+  def initialize(test_path)
+    # <GEMS_PENDING_ROOT>/spec/some_dir1/some_dir2/some_spec.rb
+    @test_file     = File.expand_path(test_path)
+
+    # <GEMS_PENDING_ROOT>/spec/some_dir1/some_dir2
+    @test_dir      = File.dirname(@test_file)
+
+    # some_spec
+    @test_basename = File.basename(@test_file, ".rb")
+
+    # some_dir1/some_dir2/some_spec
+    @relative_dir = File.join(spec_relative_path(@test_dir), @test_basename)
+
+    # <GEMS_PENDING_ROOT>/spec/recordings/some_dir1/some_dir2
+    @absolute_recording_dir = File.join(RECORDINGS_DIR, @relative_dir)
+
+    # <GEMS_PENDING_ROOT>/spec/test_env/config/some_dir1/some_dir2/some_spec.yml
+    @config_file = File.join(CONFIG_DIR, "#{@relative_dir}.yml")
+  end
+
+  # Global configuration values.
+  def self.[](tag)
+    global_config_values[tag] || global_config_filter[tag] || global_config_default[tag]
+  end
+
+  # Test-local configuration values - can override global.
+  def [](tag)
+    config_values[tag] || config_filter[tag] || config_default[tag] || self.class[tag]
+  end
+
+  #
+  # Global configuration.
+  #
+  def self.global_config
+    @global_config ||= begin
+      YAML.load_file(GLOBAL_CONFIG_FILE)
+    rescue
+      {}
+    end
+  end
+
+  def global_config
+    self.class.global_config
+  end
+
+  def self.global_config_values
+    @global_config_values ||= global_config[:values] || {}
+  end
+
+  def global_config_values
+    self.class.global_config_values
+  end
+
+  def self.global_config_filter
+    @global_config_filter ||= global_config[:filter] || {}
+  end
+
+  def global_config_filter
+    self.class.global_config_filter
+  end
+
+  def self.global_config_default
+    @global_config_default ||= global_config[:default] || {}
+  end
+
+  def global_config_default
+    @self.class.global_config_default
+  end
+
+  #
+  # Test-local configuration.
+  #
+  def config
+    @config ||= begin
+      YAML.load_file(@config_file)
+    rescue => err
+      puts err.to_s
+      {}
+    end
+  end
+
+  def config_values
+    @config_values ||= config[:values] || {}
+  end
+
+  def config_filter
+    @config_filter ||= config[:filter] || {}
+  end
+
+  def config_default
+    @config_default ||= config[:default] || {}
+  end
+
+  #
+  # Path and directory methods.
+  #
+  def cassette_for(id)
+    File.join(@relative_dir, "#{@test_basename}#{id}")
+  end
+
+  def spec_relative_path(path)
+    Pathname.new(File.expand_path(path)).relative_path_from(SPEC_PATH).to_s
+  end
+
+  def pwd_relative_path(path)
+    self.class.pwd_relative_path(path)
+  end
+
+  def self.recordings_dir
+    pwd_relative_path(RECORDINGS_DIR)
+  end
+
+  def self.pwd_relative_path(path)
+    pwd = Pathname.new(Dir.getwd)
+    Pathname.new(File.expand_path(path)).relative_path_from(pwd).to_s
+  end
+
+  def ensure_recording_dir_exists
+    FileUtils.mkdir_p(@absolute_recording_dir)
+  end
```

Comment:
Do you mind placing class methods at the top, separated from the instance methods? I know in this particular case they are fairly related (many instances reference the class), but if anything that's even harder to see the clean interface for because there are a few exceptions.